addons/portal/models/portal_mixin.py

```diff
@@ -76,3 +102,43 @@ def _notify_get_groups(self, message, groups):
         else:
             new_group = []
         return super(PortalMixin, self)._notify_get_groups(message, new_group + groups)
+
+    @api.multi
+    def get_access_action(self, access_uid=None):
```

Comment:
I have several remarks for this method. First one here !

I see you factorized it. However if you look at various old implementations there were a bit different, notably domin applied before computing the access URL. Indeed you cannot access the portal view of all records; for example sale order have to be confirmed or send to be displayed for customers. Somehow each model should be able to filter which records should go through the generic "portal access action" computation; maybe a domain, or a sub-method call. I let you check this. It should match filters applied in the portal controller; maybe it is time to factorize that in a single method for example.

Reply:
>  for example sale order have to be confirmed or send to be displayed for customers

Well, we discussed this with PO and it looked more convenient to be able to share the sale order even if it's not confirmed or send, because the customer should be able to see at any time the status of his sale order.